src/System.IO.FileSystem/src/System/IO/File.cs

```diff
@@ -725,5 +723,315 @@ public static void Decrypt(String path)
+        {
+            if (path == null)
+                throw new ArgumentNullException(nameof(path));
+            if (encoding == null)
+                throw new ArgumentNullException(nameof(encoding));
+            if (path.Length == 0)
+                throw new ArgumentException(SR.Argument_EmptyPath, nameof(path));
+
+            return cancellationToken.IsCancellationRequested
+                ? Task.FromCanceled<string[]>(cancellationToken)
+                : InternalReadAllLinesAsync(path, encoding, cancellationToken);
+        }
+
+        private static async Task<string[]> InternalReadAllLinesAsync(string path, Encoding encoding, CancellationToken cancellationToken)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(path));
+            Debug.Assert(encoding != null);
+
+            using (StreamReader sr = AsyncStreamReader(path, encoding))
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                string line;
+                List<string> lines = new List<string>();
+                while ((line = await sr.ReadLineAsync().ConfigureAwait(false)) != null)
+                {
+                    lines.Add(line);
+                    cancellationToken.ThrowIfCancellationRequested();
+                }
+
+                return lines.ToArray();
+            }
+        }
+
+        public static Task WriteAllLinesAsync(string path, IEnumerable<string> contents, CancellationToken cancellationToken = default(CancellationToken))
+            => WriteAllLinesAsync(path, contents, UTF8NoBOM, cancellationToken);
+
+        public static Task WriteAllLinesAsync(string path, IEnumerable<string> contents, Encoding encoding, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (path == null)
+                throw new ArgumentNullException(nameof(path));
+            if (contents == null)
+                throw new ArgumentNullException(nameof(contents));
+            if (encoding == null)
+                throw new ArgumentNullException(nameof(encoding));
+            if (path.Length == 0)
+                throw new ArgumentException(SR.Argument_EmptyPath, nameof(path));
+
+            return cancellationToken.IsCancellationRequested
+                ? Task.FromCanceled(cancellationToken)
+                : InternalWriteAllLinesAsync(AsyncStreamWriter(path, encoding, append: false), contents, cancellationToken);
+        }
+
+        private static async Task InternalWriteAllLinesAsync(TextWriter writer, IEnumerable<string> contents, CancellationToken cancellationToken)
+        {
+            Debug.Assert(writer != null);
+            Debug.Assert(contents != null);
+
+            using (writer)
+            {
+                foreach (string line in contents)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    // Note that this working depends on the fix to #8662, and cannot be ported without
```

Comment:
It is. This is a note in case it's ported to desktop. If this is ported and the fix for #8662 isn't then this will have a bug.